### PR TITLE
Bar Graph Colors

### PR DESCRIPTION
**Pull request by theswerd**
Add indicator on bar chart

## Screenshots

<details>
<summary>Screenshots</summary>

Screenshot 1
Screenshot 2
Screenshot 3
Screenshot 4
Demo

</details>

## How did you test the change?

- [x] iOS Simulator
- [ ] iOS Device
- [x] Android Simulator
- [ ] Android Device
- [ ] `curl` to a dev App Engine server
- [ ] other, please describe

## Checklist:

- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md) and verified that all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE).

### CODE REVIEW

File: client/lib/components/recent_numbers_graph.dart [hunk 1]
```diff
@@ -40,6 +54,19 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
           show: false,
         ),
         barTouchData: BarTouchData(
+          touchCallback: (touchResponse) {
+            setState(() {
+              swapDuration = Duration(microseconds: 1);
```
Comment (brunobowden) [inline client/lib/components/recent_numbers_graph.dart:59]: Why 1000 Millisecs here and 750 above? Ideally the duration should be specified once only.

More importantly, why change the value at all? I’d expect there to be a single default and not to need changing it anywhere as you also do below.

https://en.m.wikipedia.org/wiki/Don%27t_repeat_yourself

File: client/lib/components/recent_numbers_graph.dart [hunk 2]
```diff
@@ -40,6 +54,19 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
...
+              if (touchResponse.spot != null &&
+                  touchResponse.touchInput is! FlPanEnd &&
+                  touchResponse.touchInput is! FlLongPressEnd) {
+                selectedIndex = touchResponse.spot.touchedRodDataIndex;
+              } else {
+                selectedIndex = null;
+                swapDuration = Duration(milliseconds: 750);
```
Comment (brunobowden) [inline client/lib/components/recent_numbers_graph.dart:66]: Should be using default and not need adjustment

File: client/lib/components/recent_numbers_graph.dart [hunk 3]
```diff
@@ -109,7 +140,9 @@ class _RecentNumbersBarGraphState extends State<RecentNumbersBarGraph> {
         bars.add(
           BarChartRodData(
             y: 0,
-            color: Constants.textColor.withOpacity(.3),
+            color: Constants.textColor.withOpacity(
+              .3,
+            ),
```
Comment (brunobowden) [inline client/lib/components/recent_numbers_graph.dart:145]: Style change above is allowed due to editing code. This section shouldn’t be changed as you’re not changing anything.

**Review by brunobowden — APPROVED** [review]
Thanks Ben. The animated screenshot is very helpful.

This is optional but should the colored bar be a little wider? How visible is it right now on small mobile devices? I don’t have a strong opinion on this but something we should consider in QA. FYI @Mohit020888

File: client/lib/components/recent_numbers_graph.dart (hunk 2 above)
Comment (theswerd) [reply client/lib/components/recent_numbers_graph.dart:66]: The duration for animating the swap between **Daily** and **Total Stats** is 750 milliseconds, however the animation for changing the bars should be instantaneous. This is here so after the bar color changes are done, the animation returns to its original state.

File: client/lib/components/recent_numbers_graph.dart (hunk 3 above)
Comment (theswerd) [reply client/lib/components/recent_numbers_graph.dart:145]: Originally had a change on that line -- will revert now

File: client/lib/components/recent_numbers_graph.dart (hunk 1 above)
Comment (brunobowden) [reply client/lib/components/recent_numbers_graph.dart:59]: Sorry, misread that... it's 0.001 milliseconds for a microsecond. That's fine then.

**Review by brunobowden — COMMENTED** [review]
Extra 750 milliseconds to shared constant?

**Comment by theswerd** [comment]
@brunobowden done -- 750 milliseconds has been refactored

**Comment by brunobowden** [comment]
@theswerd - looks good, merge at your leisure

**Pull request merged**